### PR TITLE
Use subprocesses in `TestCheckSpecs` and `TestScaffoldFSMAutonomyTests`

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

This PR makes `TestCheckSpecs` and `TestScaffoldFSMAutonomyTests` to use subprocess for each test to better isolate the Python environment of the CLI command being executed.

## Fixes

n/a

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [X] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

n/a

### CODE REVIEW

File: tests/test_autonomy/test_cli/test_analyse/test_abci/test_specs.py [hunk 1]
```diff
@@ -229,6 +231,12 @@ def test_failures(
             "Please provide path to specification file." in result.output
         ), result.output
 
+    def teardown(
+        self,
+    ) -> None:
+        """Tear down the test."""
```
Comment (DavidMinarsch) [inline tests/test_autonomy/test_cli/test_analyse/test_abci/test_specs.py:237]: you need to call `super().teardown()` o/w we don't have proper cleanup

Comment (marcofavorito) [reply tests/test_autonomy/test_cli/test_analyse/test_abci/test_specs.py:237]: right

**Review by Karrenbelt — COMMENTED** [review]
I think pytest monkeypatch might offer a solution: https://docs.pytest.org/en/7.1.x/how-to/monkeypatch.html

**Comment by marcofavorito** [comment]
> I think pytest monkeypatch might offer a solution: https://docs.pytest.org/en/7.1.x/how-to/monkeypatch.html

Thank you for the suggestion! I will use it.

**Comment by marcofavorito** [comment]
Finally tests are passing. Squashing all commits into one.

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**